dsgrid/dataset/dataset_schema_handler_standard.py

```diff
@@ -40,10 +51,123 @@ def get_unique_dimension_rows(self):
         """Get distinct combinations of remapped dimensions, including id.
         Check each col in combination for null value."""
         dim_table = self._remap_dimension_columns(self._load_data_lookup).distinct()
-        self._check_null_value_in_unique_dimension_rows(dim_table)
-
+        check_null_value_in_unique_dimension_rows(dim_table)
         return dim_table
 
+    def make_project_dataframe(
+        self, load_data_df=None, lookup_df=None, query_context: QueryContext = None
+    ):
+        ld_df = load_data_df or self._load_data
+        lk_df = lookup_df or self._load_data_lookup
+
+        if query_context is not None:
+            lk_df, ld_df = self._prefilter_dataset(query_context, lk_df, ld_df)
+
+        lk_df = self._remap_dimension_columns(lk_df)
+        ld_df = self._remap_dimension_columns(
+            ld_df,
+            # Some pivot columns may have been removed.
+            pivot_columns=set(ld_df.columns).intersection(self.get_pivot_dimension_columns()),
+        )
+
+        if query_context is not None:
+            ld_df = self._process_metric_aggregations(query_context, ld_df)
+
+        ld_df = ld_df.join(lk_df.drop("fraction"), on="id").drop("id")
+
+        # Map the time here because some columns may have been collapsed above.
+        ld_df = self._convert_time_dimension(ld_df)
+
+        # All dataset columns need to be in the same order.
+        pivot_columns = sorted(
+            set(ld_df.columns).intersection(self.get_pivot_dimension_columns_mapped_to_project())
+        )
+        dim_columns = sorted(set(ld_df.columns).intersection([x.value for x in DimensionType]))
+        covered = set(pivot_columns + dim_columns)
+        remaining = sorted((x for x in ld_df.columns if x not in covered))
+        final_columns = remaining + dim_columns + pivot_columns
+        return ld_df.select(*final_columns)
+
+    def _prefilter_dataset(self, context: QueryContext, lk_df, ld_df):
+        # TODO: prefilter time
+        if context.model.project.drop_dimensions:
+            columns = [x.value for x in context.model.project.drop_dimensions]
+            lk_df = lk_df.drop(*columns)
+
+        for dim_type, project_record_ids in context.iter_record_ids_by_dimension_type():
+            dataset_mapping = self._get_dataset_to_project_mapping_records(dim_type)
```

Comment:
Does this only return the simple downselection mappings without multipliers (i.e. with default from fractions) ?

Reply:
There is no mapping here. It should only remove rows or columns that the project is not asking for.